orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/tasks/MonitorFront50Task.java

```diff
@@ -114,22 +97,76 @@ public TaskResult execute(@Nonnull Stage stage) {
         );
         return TaskResult.RUNNING;
       }
+    } else if (stageData.deliveryConfig != null) {
+      String deliveryConfigId = stageData.deliveryConfig.getId();
+      try {
+        return monitor(this::getDeliveryConfig, deliveryConfigId, stage.getStartTime());
+      } catch (Exception e) {
+        log.error(
+          "Unable to verify that delivery config has been updated (executionId: {}, configId: {})",
+          stage.getExecution().getId(),
+          deliveryConfigId,
+          e
+        );
+        return TaskResult.RUNNING;
+      }
     } else {
       log.warn(
-        "No pipeline id found, unable to verify that pipeline has been updated (executionId: {}, pipeline: {})",
-        stage.getExecution().getId(),
-        stageData.pipelineName
+        "No id found, unable to verify that the object has been updated (executionId: {})",
+        stage.getExecution().getId()
       );
     }
 
     return new TaskResult(ExecutionStatus.SUCCEEDED);
   }
 
+  private TaskResult monitor(Function<String, Optional<Map<String,Object>>> getObjectFunction, String id, Long startTime) {
+    /*
+     * Some storage services (notably S3) are eventually consistent when versioning is enabled.
+     *
+     * This "dirty hack" attempts to ensure that each underlying instance of Front50 has cached an _updated copy_
+     * of the modified resource.
+     *
+     * It does so by making multiple requests (currently only applies to pipelines) with the expectation that they
+     * will round-robin across all instances of Front50.
+     */
+    for (int i = 0; i < successThreshold; i++) {
+      Optional<Map<String, Object>> object = getObjectFunction.apply(id);
+      if (!object.isPresent()) {
+        return TaskResult.RUNNING;
+      }
+
+      Long lastModifiedTime;
+      if (object.get().containsKey("updateTs")) {
+        lastModifiedTime = Long.valueOf(object.get().get("updateTs").toString());
+      } else {
+        lastModifiedTime = Long.valueOf(object.get().get("lastModified").toString());
+      }
+
+      if (lastModifiedTime < (startTime - gracePeriodMs)) {
```

Comment:
speaking out loud: seems super hoaky because it relies on the various service instance to have reasonably synchronized timestamps... (e.g. `startTime` is set by one instance, `updateTs` by another, etc)

Reply:
This is not my code or my "dirty hack", I'm just generalizing it to work for monitoring other front50 objects.

Reply:
We're all ntp'd, so timestamp drift should be quite minimal. It's not performing precise calculations here, so I think it's "fine" as is. The better way would be to have a deadline timestamp, but meh.